### ISSUE TITLE
Appium in Yosemite - instruments WebKit Threading Violation - initial use of WebKit from a secondary thread.

### ISSUE THREAD

**Issue by bhawani2008** [appium/appium#4637]
Hi,
I am working with Appium@1.3.4 and Protractor for iOS hybrid app automation in Yosemite. I am getting the error on executing the protractor tests with Appium server.
Kindly advise.

My Config file:
exports.config = {
  allScriptsTimeout: 50000 ,
  getPageTimeout: 50000 ,
  seleniumAddress: 'http://localhost:4723/wd/hub',
  
	specs: ['PageObjectSpec.js'],

 capabilities: {
     browserName: 'iOS' ,
    'appium-version':'1.3.4',
     platformName:'iOS'   ,
     platformVersion: 8.1 ,
     deviceName:'iPhone6' ,
     autoWebview:true , 
     app: /Users/qualcomm/Desktop/bhawani/AbbVieCareCH.app ,
  },

   baseUrl: 'http://10.0.2.2:' + (process.env.HTTP_PORT || '8000') ,
   jasmineNodeOpts: {showColors: true, defaultTimeoutInterval: 50000} ,
};   

APPIUM LOGS::

info: Welcome to Appium v1.3.4 (REV c8c79a85fbd6870cd6fc3d66d038a115ebe22efe)
info: Appium REST http interface listener started on 0.0.0.0:4723
info: [debug] Non-default server args: {"app":"/Users/Desktop/bhawani/AbbVieCareCH.app","ipa":"/Users/Desktop/bhawani/abbvieCH-dev-01.01.21050131.01.ipa","udid":"cbecb05c855c13d9b2e6b693999123268fc0b966"}
info: Console LogLevel: debug
info: --> POST /wd/hub/session {"desiredCapabilities":{"browserName":"iOS","appium-version":"1.3.4","platformName":"iOS","platformVersion":8.1,"deviceName":"iPhone6","autoWebview":true,"count":1}}
info: Client User-Agent string: undefined
info: [debug] The following desired capabilities were provided, but not recognized by appium. They will be passed on to any other services running on this server. : appium-version, count
info: [debug] Using local app from command line: /Users/Desktop/bhawani/AbbVieCareCH.app
info: [debug] Creating new appium session f9bf457f-1cd1-4638-8db8-41dd69d82dee
info: [debug] Removing any remaining instruments sockets
info: [debug] Cleaned up instruments socket /tmp/instruments_sock
info: [debug] Setting Xcode folder
info: [debug] Setting Xcode version
info: [debug] Setting iOS SDK Version
info: [debug] Getting sdk version from xcrun with a timeout
info: [debug] iOS SDK Version set to 8.1
info: [debug] Not checking whether simulator is available since we're on a real device
info: [debug] Detecting automation tracetemplate
info: [debug] Not auto-detecting udid, running on sim
info: [debug] Parsed app Info.plist (as binary)
info: [debug] Parsed app Localizable.strings
info: [debug] Getting bundle ID from app
info: [debug] Parsed app Info.plist (as binary)
info: [debug] Creating instruments
info: On some xcode 6 platforms, instruments-without-delay does not work. If you experience this, you will need to re-run appium with the --native-instruments-lib flag
info: [debug] Preparing uiauto bootstrap
info: [debug] Dynamic bootstrap dir: /Users/Library/Application Support/appium/bootstrap
info: [debug] Dynamic env: {"nodePath":"/usr/local/bin/node","commandProxyClientPath":"/usr/local/lib/node_modules/appium/node_modules/appium-uiauto/bin/command-proxy-client.js","instrumentsSock":"/tmp/instruments_sock","interKeyDelay":null,"justLoopInfinitely":false,"autoAcceptAlerts":false,"sendKeyStrategy":"grouped"}
info: [debug] Dynamic bootstrap code: // This file is automatically generated. D...ot manually modify!
info: [debug] Dynamic bootstrap path: /Users//Library/Application Support/appium/bootstrap/bootstrap-d0f21a5b61bb08af.js
info: [debug] Reusing dynamic bootstrap: /Users/Library/Application Support/appium/bootstrap/bootstrap-d0f21a5b61bb08af.js
info: [debug] Getting device string from opts: {"forceIphone":false,"forceIpad":false,"xcodeVersion":"6.1.1","iOSSDKVersion":"8.1","deviceName":"iPhone6","platformVersion":8.1}
info: [debug] fixDevice is on
info: [debug] Final device string is: 'iPhone6 (8.1 Simulator)'
info: [debug] Not setting device type since we're on a real device
info: [debug] Checking whether we need to set app preferences
info: [debug] Not setting iOS and app preferences since we're on a real device
info: [debug] Running ios sim reset flow
info: [debug] Killing the simulator process
info: [debug] Killall iOS Simulator
info: [debug] Killing any other simulator daemons
info: [debug] On a real device; cannot clean device state
info: [debug] Not setting locale because we're using a real device
info: [debug] No iOS / app preferences to set
info: [debug] Starting iOS device log capture via deviceconsole
info: [debug] Not pre-launching simulator
info: [debug] Creating iDevice object with udid cbecb05c855c13d9b2e6b693999123268fc0b966
info: [debug] Checking app install status using: /usr/local/lib/node_modules/appium/build/fruitstrap/fruitstrap isInstalled --id cbecb05c855c13d9b2e6b693999123268fc0b966 --bundle com.qcl.healthycircles.dev
info: [debug] App is installed.
info: [debug] fullReset not requested. No need to install.
info: [debug] Starting command proxy.
info: [debug] Instruments socket server started at /tmp/instruments_sock
info: [debug] Starting instruments
info: [debug] Instruments is at: /Applications/Xcode.app/Contents/Developer/usr/bin/instruments
info: Launching instruments
info: [debug] Attempting to run app on real device with UDID cbecb05c855c13d9b2e6b693999123268fc0b966
info: [debug] Spawning instruments with command: /Applications/Xcode.app/Contents/Developer/usr/bin/instruments -t /Applications/Xcode.app/Contents/Applications/Instruments.app/Contents/PlugIns/AutomationInstrument.xrplugin/Contents/Resources/Automation.tracetemplate -D /tmp/appium-instruments/instrumentscli0.trace -w cbecb05c855c13d9b2e6b693999123268fc0b966 com.qcl.healthycircles.dev -e UIASCRIPT "/Users/Library/Application Support/appium/bootstrap/bootstrap-d0f21a5b61bb08af.js" -e UIARESULTSPATH /tmp/appium-instruments
info: [debug] And extra without-delay env: {}
info: [debug] And launch timeouts (in ms): {"global":90000}
info: [debug] [INST STDERR] 2015-03-02 18:32:26.224 instruments[12949:128680] WebKit Threading Violation - initial use of WebKit from a secondary thread.
info: [debug] [INST STDERR] Instruments Trace Error : Target failed to run: Permission to debug com.qcl.healthycircles.dev was denied. The app must be signed with a development identity (e.g. iOS Developer).
info: [debug] [INSTSERVER] Instruments exited with code 253
info: [debug] Killall instruments
info: [debug] Instruments crashed on startup
info: [debug] Attempting to retry launching instruments, this is retry #1
info: [debug] Killall iOS Simulator
info: Launching instruments
info: [debug] Attempting to run app on real device with UDID cbecb05c855c13d9b2e6b693999123268fc0b966
info: [debug] Spawning instruments with command: /Applications/Xcode.app/Contents/Developer/usr/bin/instruments -t /Applications/Xcode.app/Contents/Applications/Instruments.app/Contents/PlugIns/AutomationInstrument.xrplugin/Contents/Resources/Automation.tracetemplate -D /tmp/appium-instruments/instrumentscli0.trace -w cbecb05c855c13d9b2e6b693999123268fc0b966 com.qcl.healthycircles.dev -e UIASCRIPT "/Users/Library/Application Support/appium/bootstrap/bootstrap-d0f21a5b61bb08af.js" -e UIARESULTSPATH /tmp/appium-instruments
info: [debug] And extra without-delay env: {}
info: [debug] And launch timeouts (in ms): {"global":90000}
info: [debug] [INST STDERR] 2015-03-02 18:32:33.470 instruments[12952:128775] Connection peer refused channel request for "com.apple.instruments.server.services.launchdaemon"; channel canceled <DTXChannel: 0x7fdbc06a32c0>
info: [debug] [INST STDERR] 2015-03-02 18:32:33.472 instruments[12952:128776] Connection peer refused channel request for "com.apple.instruments.server.services.device.xpccontrol"; channel canceled <DTXChannel: 0x7fdbc1d8fc30>
info: [debug] [INST STDERR] 2015-03-02 18:32:33.474 instruments[12952:128775] Connection peer refused channel request for "com.apple.instruments.server.services.deviceinfo"; channel canceled <DTXChannel: 0x7fdbc1a95420>
info: [debug] [INST STDERR] 2015-03-02 18:32:33.475 instruments[12952:128776] Connection peer refused channel request for "com.apple.instruments.server.services.processcontrol"; channel canceled <DTXChannel: 0x7fdbc1d8e490>
info: [debug] [INST STDERR] 2015-03-02 18:32:33.476 instruments[12952:128775] Connection peer refused channel request for "com.apple.instruments.server.services.processcontrol.posixspawn"; channel canceled <DTXChannel: 0x7fdbc1a94f10>
info: [debug] [INST STDERR] 2015-03-02 18:32:33.477 instruments[12952:128776] Connection peer refused channel request for "com.apple.instruments.server.services.mobilenotifications"; channel canceled <DTXChannel: 0x7fdbc041b990>
info: [debug] [INST STDERR] 2015-03-02 18:32:33.479 instruments[12952:128776] Connection peer refused channel request for "com.apple.instruments.server.services.device.applictionListing"; channel canceled <DTXChannel: 0x7fdbc0420330>
info: [debug] [INST STDERR] 2015-03-02 18:32:33.506 instruments[12952:128781] Connection peer refused channel request for "com.apple.instruments.server.services.wireless"; channel canceled <DTXChannel: 0x7fdbc05034e0>
info: [debug] [INST STDERR] 2015-03-02 18:32:33.523 instruments[12952:128775] WebKit Threading Violation - initial use of WebKit from a secondary thread.
info: [debug] [INST STDERR] Instruments Trace Error : Target failed to run: Permission to debug com.qcl.healthycircles.dev was denied. The app must be signed with a development identity (e.g. iOS Developer).
info: [debug] [INSTSERVER] Instruments exited with code 253
info: [debug] Killall instruments
info: [debug] Instruments crashed on startup
info: [debug] Attempting to retry launching instruments, this is retry #2
info: [debug] Killall iOS Simulator
info: Launching instruments
info: [debug] Attempting to run app on real device with UDID cbecb05c855c13d9b2e6b693999123268fc0b966
info: [debug] Spawning instruments with command: /Applications/Xcode.app/Contents/Developer/usr/bin/instruments -t /Applications/Xcode.app/Contents/Applications/Instruments.app/Contents/PlugIns/AutomationInstrument.xrplugin/Contents/Resources/Automation.tracetemplate -D /tmp/appium-instruments/instrumentscli0.trace -w cbecb05c855c13d9b2e6b693999123268fc0b966 com.qcl.healthycircles.dev -e UIASCRIPT "/Users/Library/Application Support/appium/bootstrap/bootstrap-d0f21a5b61bb08af.js" -e UIARESULTSPATH /tmp/appium-instruments
info: [debug] And extra without-delay env: {}
info: [debug] And launch timeouts (in ms): {"global":90000}
info: [debug] [INST STDERR] 2015-03-02 18:32:39.919 instruments[12955:128821] WebKit Threading Violation - initial use of WebKit from a secondary thread.
info: [debug] [INST STDERR] Instruments Trace Error : Target failed to run: Permission to debug com.qcl.healthycircles.dev was denied. The app must be signed with a development identity (e.g. iOS Developer).
info: [debug] [INSTSERVER] Instruments exited with code 253
info: [debug] Killall instruments
info: [debug] Instruments crashed on startup
info: [debug] Attempting to retry launching instruments, this is retry #3
info: [debug] Killall iOS Simulator
info: Launching instruments
info: [debug] Attempting to run app on real device with UDID cbecb05c855c13d9b2e6b693999123268fc0b966
info: [debug] Spawning instruments with command: /Applications/Xcode.app/Contents/Developer/usr/bin/instruments -t /Applications/Xcode.app/Contents/Applications/Instruments.app/Contents/PlugIns/AutomationInstrument.xrplugin/Contents/Resources/Automation.tracetemplate -D /tmp/appium-instruments/instrumentscli0.trace -w cbecb05c855c13d9b2e6b693999123268fc0b966 com.qcl.healthycircles.dev -e UIASCRIPT "/Users/Library/Application Support/appium/bootstrap/bootstrap-d0f21a5b61bb08af.js" -e UIARESULTSPATH /tmp/appium-instruments
info: [debug] And extra without-delay env: {}
info: [debug] And launch timeouts (in ms): {"global":90000}
info: [debug] [INST STDERR] 2015-03-02 18:32:46.162 instruments[12958:128870] WebKit Threading Violation - initial use of WebKit from a secondary thread.
info: [debug] [INST STDERR] Instruments Trace Error : Target failed to run: Permission to debug com.qcl.healthycircles.dev was denied. The app must be signed with a development identity (e.g. iOS Developer).
info: [debug] [INSTSERVER] Instruments exited with code 253
info: [debug] Killall instruments
info: [debug] Instruments crashed on startup
info: [debug] We exceeded the number of retries allowed for instruments to successfully start; failing launch
info: [debug] Stopping iOS log capture
info: [debug] Running ios sim reset flow
info: [debug] Killing the simulator process
info: [debug] Killall iOS Simulator
info: [debug] Killing any other simulator daemons
info: [debug] On a real device; cannot clean device state
info: [debug] Cleaning up appium session
error: Failed to start an Appium session, err was: Error: Instruments crashed on startup
info: [debug] Error: Instruments crashed on startup
    at Instruments.onInstrumentsExit (/usr/local/lib/node_modules/appium/node_modules/appium-instruments/lib/instruments.js:387:31)
    at null.<anonymous> (/usr/local/lib/node_modules/appium/node_modules/appium-instruments/lib/instruments.js:308:12)
    at ChildProcess.emit (events.js:98:17)
    at Process.ChildProcess._handle.onexit (child_process.js:810:12)
info: [debug] Responding to client with error: {"status":33,"value":{"message":"A new session could not be created. (Original error: Instruments crashed on startup)","origValue":"Instruments crashed on startup"},"sessionId":null}
info: <-- POST /wd/hub/session 500 25403.185 ms - 182

**Comment by imurchie** [comment]
```
info: [debug] [INST STDERR] Instruments Trace Error : Target failed to run: Permission to debug com.qcl.healthycircles.dev was denied. The app must be signed with a development identity (e.g. iOS Developer).
```

Is the `ipa` you are trying to automate signed with a development identity? See [here](https://github.com/appium/appium/blob/master/docs/en/appium-setup/real-devices.md#appium-on-real-ios-devices).

**Comment by bhawani2008** [comment]
Thanks a lot!! imurchie for the quick response. My ipa is built with enterprise certificate and not with developer's certificate. I will get in touch with the developer for the same....

**Comment by sebv** [comment]
Has this been sorted?

**Comment by mrudulp** [comment]
Just as a referencing tip, in case someone has similar troubles, but not a result of certificate issues,

Verify "UI Automation" is **enabled** on the device.
For me similar error of "Webkit violation" was a result of "UI Automation" not enabled.

To enable UI Automation --
* Goto Settings on device
* Scroll to "Developer" option
* Switch on the option for UI Automation

**Comment by sebv** [comment]
Ok closing, sound like it is not a bug. Open a new issue if it happens again.

**Comment by reshmiTechm** [comment]
I am also getting the same error. I have enabled UI Automation on the device but still the error is observed. Can someone please help.

**Comment by wkoszek** [comment]
@sebv @mrudulp  I don't know what the implication of this message is, but with iOS 9.2.1 and Appium `[Appium] Welcome to Appium v1.5.1 (REV bfb83f5033d24b3daff49cbe81e9c4187b7bc547)` it's still showing up o El Captain.

**Comment by aav789** [comment]
appium -v   1.5.1
node -v   v5.10.0
iOS 9.0
Mac OS 10.11.4

Command 'instruments -s devices'  shows my device attached

I am also getting the error when i run:
instruments -t "/Applications/Xcode.app/Contents/Applications/Instruments.app/Contents/PlugIns/AutomationInstrument.xrplugin/Contents/Resources/Automation.tracetemplate" -w "M3_iPhone6_1 (9.0)" "/usr/local/lib/node_modules/appium/build/SafariLauncher/SafariLauncher.app"
2016-04-04 16:32:01.171 instruments[62479:2812713] WebKit Threading Violation - initial use of WebKit from a secondary thread.

**Comment by imurchie** [comment]
That should not affect the operation of Appium or Instruments.

**Comment by mrudulp** [comment]
@wkoszek are you able to run UIAutomation tool ? 

I had the same problem recently when I upgraded to new mac. Noticed that UIAutomation was not able to run as well.
Solution to it was to open Xcode and let it extract symbols from the device under test.
UIAutomation will prompt for the same, indicating problem and its solution
hope it helps

**Comment by wkoszek** [comment]
@mrudulp I'm seeing it all the time with all sort of appium versions.

I've had many problems with `1.5.1` and `1.4.16`. The only `appium` which runs well for both simulator and a device is `1.4.14` installed via `nvm`.

But with 1.4.14 I'm seeing this too.

**Comment by aroramayank2002** [comment]
I got this issue when trying to run tests parallel on two iOS devices.

**Comment by Alhajsalem** [comment]
I have the same problem when I am trying to run tests in parallel on two different real iOS devices, any solutions?

**Comment by aroramayank2002** [comment]
Resolved for me using following configurations, I am using WDIO. This seems to be the only thread on  internet where useful information is provided for this issue.

Device 1:
npm run wdio – conf/a4e8adc16dbb7e2e2a72f6aa85670882515da8ea.js
ios_webkit_debug_proxy -c a4e8adc16dbb7e2e2a72f6aa856702515da8ea:9223 -d > /Users/Shared/temp/log0
appium -p 4724 -bp 4734 --session-override --webkit-debug-proxy-port 9223 -U a4e8adc16dbb7e2e2a72f6aa85670882515da8ea --app ‘safari’ --tmp /Users/Shared/temp/tmp0

Device 2:
npm run wdio – conf/47e8c20aba1aee417aac7310bec11ab87884a79d.js
ios_webkit_debug_proxy -c 47e8c20aba1aee417aac7310bec1187884a79d:9222 -d > /Users/Shared/temp/log1
appium -p 4725 -bp 4735 --webkit-debug-proxy-port 9222 -U a4e8adc16dbb7e2e2a72f6aa85670882515da8ea --app 'safari' --session-override --tmp /Users/Shared/temp/tmp1

**Comment by siva2503** [comment]
If the device's IOS version is 9.3.4  there is a bug raised already -https://github.com/appium/appium/issues/6730.
